lib/ruby-debug-ide/xml_printer.rb

```diff
@@ -380,13 +387,57 @@ def max_compact_name_size
       50
     end
 
+    def inspect_with_allocation_control(slice, memory_limit)
+      curr_thread = Thread.current
+      result = nil
+      inspect_thread = DebugThread.start {
+        start_alloc_size = ObjectSpace.memsize_of_all
+        start_time = Time.now
+        max_time = Debugger.evaluation_timeout
```

Comment:
I think 10s evaluation timeout is too much for this place. The reason is that evaluation is an explicit user action so it's OK to try to wait for a possibly long evaluation. In the case of convenience infos calculated dozens of times per breakpoint, 10s does not make much sense to me. I'd say we should make it somewhere between 100 and 500ms.